src/DotNet.Status.Web/Models/IssueMentionForwarder.cs

```diff
@@ -0,0 +1,97 @@
+    }
+
+    public class IssueMentionForwarder : IIssueMentionForwarder
+    {
+        private readonly IOptionsSnapshot<IssueMentionForwardingOptions> _options;
+        private readonly ILogger<IssueMentionForwarder> _logger;
+        private readonly IHttpClientFactory _httpClientFactory;
+
+        public IssueMentionForwarder(IHttpClientFactory httpClientFactory, IOptionsSnapshot<IssueMentionForwardingOptions> options, ILogger<IssueMentionForwarder> logger)
+        {
+            _httpClientFactory = httpClientFactory;
+            _options = options;
+            _logger = logger;
+        }
+
+        public async Task<bool> HandleIssueBody([CanBeNull] string oldBody, string newBody, string title, string commentUri, string username, DateTimeOffset date)
+        {
+            var teams = _options.Value.WatchedTeams;
+            var foundTeam = teams.FirstOrDefault(newBody.Contains);
+            if (string.IsNullOrEmpty(foundTeam))
+            {
+                return false;
+            }
+            if (!string.IsNullOrEmpty(oldBody))
+            {
+                var oldTeam = teams.FirstOrDefault(oldBody.Contains);
+                if (oldTeam == foundTeam)
+                {
+                    return false;
+                }
+            }
+
+            await SendCommentNotification(title, newBody, foundTeam, commentUri, username, date);
+            return true;
+        }
+
+        private async Task SendCommentNotification(string title, string body, string teamName, string commentUri, string username, DateTimeOffset date)
+        {
+            var message = new MessageCard
+            {
+                ThemeColor = "0072C6",
+                Text = $"Team {teamName} was mentioned in an issue.",
+                Actions = new List<IAction> {
+                    new OpenUri {
+                        Name = "Open comment",
+                        Targets = new List<Target>
+                        {
+                            new Target
+                            {
+                                OperatingSystem = "default",
+                                Uri = commentUri
+                            }
+                        }
+                    }
+                },
+                Sections = new List<Section> {
+                    new Section {
+                        ActivityTitle = title,
+                        ActivitySubtitle = $"{username} on {date:g}",
+                        ActivityText = body
+                    }
+                }
+            };
+            HttpClient client = _httpClientFactory.CreateClient();
+            using var req = new HttpRequestMessage(HttpMethod.Post, _options.Value.TeamsWebHookUri)
+            {
+                Content = new StringContent(JsonConvert.SerializeObject(message), Encoding.UTF8),
+            };
+            using var res = await client.SendAsync(req);
+            res.EnsureSuccessStatusCode();
```

Comment:
It's odd for an interface to throw such an uncontrolled exception as this. :-)  This means the caller of "IMentionHandler" has to understand the internals of all implementations of the interface, which is a pretty leaky abstraction.

If we want to throw something, we should throw a specific kind of exception.

Reply:
that makes sense, I will fix it.